validations/src/validations.rs

```diff
@@ -398,42 +398,58 @@ pub fn validate_tally_transaction(
         Err(TransactionError::InvalidTallyTransaction)?
     }
 }
-/// Function to validate a signature
-pub fn validate_transaction_signature(
-    input: &Input,
-    keyed_signature: &KeyedSignature,
-) -> Result<(), failure::Error> {
-    let tx_hash = match input {
-        Input::Commit(i) => i.transaction_id,
-        Input::Reveal(i) => i.transaction_id,
-        Input::ValueTransfer(i) => i.transaction_id,
-        Input::DataRequest(i) => i.transaction_id,
-    };
 
-    let Hash::SHA256(message) = tx_hash;
+/// Function to validate a block signature
+pub fn validate_block_signature(block: &Block) -> Result<(), failure::Error> {
+    let keyed_signature = &block.proof.block_sig;
 
-    let signature = match keyed_signature.signature.clone() {
-        Signature::Secp256k1(s) => s.try_into()?,
-    };
+    let signature = keyed_signature.signature.clone().try_into()?;
     let public_key = keyed_signature.public_key.clone().try_into()?;
 
-    verify(public_key, &message, signature)
+    let Hash::SHA256(message) = block.block_header.beacon.hash();
+
+    verify(&public_key, &message, &signature)
+        .map_err(|_| BlockError::VerifySignatureFail { hash: block.hash() }.into())
 }
 
-/// Function to validate signatures
-pub fn validate_transaction_signatures(
-    inputs: Vec<Input>,
-    signatures: Vec<KeyedSignature>,
+/// Function to validate a pkh signature
+pub fn validate_pkh_signature(
+    _input: &Input,
+    _keyed_signature: &KeyedSignature,
 ) -> Result<(), failure::Error> {
+    // TODO: Implement a properly pkh validation
+    Ok(())
+}
+
+/// Function to validate transaction signatures
+pub fn validate_transaction_signatures(transaction: &Transaction) -> Result<(), failure::Error> {
+    let signatures = &transaction.signatures;
+    let inputs = &transaction.body.inputs;
+
     if signatures.len() != inputs.len() {
         Err(TransactionError::MismatchingSignaturesNumber {
             signatures_n: signatures.len() as u8,
             inputs_n: inputs.len() as u8,
         })?
     }
 
-    for (input, keyed_signature) in inputs.iter().zip(signatures.iter()) {
-        validate_transaction_signature(input, keyed_signature)?
+    // Validate transaction signature
+    if let Some(tx_keyed_signature) = signatures.get(0) {
+        let signature = tx_keyed_signature.signature.clone().try_into()?;
+        let public_key = tx_keyed_signature.public_key.clone().try_into()?;
+        let Hash::SHA256(message) = transaction.hash();
+
+        verify(&public_key, &message, &signature).map_err(|_| {
+            TransactionError::VerifySignatureFail {
+                hash: transaction.hash(),
+            }
+        })?;
+    } else {
+        Err(TransactionError::SignatureNotFound)?
+    }
+
+    for (input, keyed_signature) in inputs.iter().zip(signatures.iter().next()) {
+        validate_pkh_signature(input, keyed_signature)?
```

Comment:
This also needs to be passed the transaction or at least the transaction's hash (in terms of performance is the same as the hash is memoized, but IMO it's always better to operate on the minimum set of data possible, so I'd prefer the latter)